### DISCUSSION TITLE
How to make My Candidates generated by Acquisition function always lie on a simplex ?

### DISCUSSION THREAD

**Discussion by sheikhahnaf** [meta-pytorch/botorch#2430] [Q&A]
Hi,
I was wondering, **How to make My Candidates generated by Acquisition function always lie on a simplex ?**. Is it through equality constraints passed to the optimize_acqf? where, for example, for 3 dimension the equality constraints should be
 [(tensor([1,2,3]),tensor([1,1,1]),1.0)]? or are there any other ways? I tried this for 1 dimension and it always generates candidates with value 1, so I guess it will work for the higher dimensions. 

1.I was looking any better method, because I saw a function in botorch.utils.optim named sample_simplex, but couldn't find any sampler where it is used. 
2.Moreover, what if I want my candidates to be drawn from a specific distribution like, Dirichlet distribution(which samples from a simplex uniformly)? How do I go about implementing that?

Thanks in Advance

**Comment by esantorella** [comment]
Yes, passing equality constraints to `optimize_acqf` would be a good way of doing this if you want to do Bayesian Optimization with a surrogate model and an acquisition function. `sample_simplex` would work for IID uniform sampling.
